samples/cpp/classification_sample_async/main.cpp

```
@@ -170,7 +170,7 @@ int main(int argc, char* argv[]) {
         // -------- Step 10. Do asynchronous inference --------
         infer_request.set_callback([&](std::exception_ptr ex) {
             if (ex)
-                throw ex;
+                std::rethrow_exception(ex);
```

You need to propagate the exception to the main thread

---

@Wovchena could you suggest what exactly should be done?

---

Same try catch as in https://github.com/openvinotoolkit/open_model_zoo/pull/3376

---

Please use `std::promise<void>/std::future<void>` to wait for execution or propagate exception to main thread. 
Please use `OVInferRequestCallbackTests.canStartSeveralAsyncInsideCompletionCallbackWithSafeDtor` as example

---

@apankratovantonp , please see current variant using variable for checking status of exception. Sample gets error and not hanging without std::future

---

@apankratovantonp or do you want to realize it using std::future instead of current solution?

---

The last solution is enough to propagate exception to main thread.